frost-coordinator/src/coordinator.rs

```diff
@@ -469,51 +469,131 @@ where
 
 #[cfg(test)]
 mod test {
+    use crate::{
+        coordinator::{Command, Coordinator},
+        DEVNET_COORDINATOR_ID,
+    };
+    use frost_signer::{
+        config::{Config, PublicKeys, SignerKeyIds},
+        logging,
+        net::{HttpNet, HttpNetListen},
+        signer::Signer,
+    };
     use hashbrown::HashMap;
-    use test_utils::Process;
+    use p256k1::{ecdsa, scalar::Scalar};
+    use rand_core::OsRng;
+    use relay_server::Server as RelayServer;
+    use std::{env, thread};
+
+    fn create_signer_key_ids(signer_id: u32, keys_per_signer: u32) -> Vec<u32> {
+        let mut signer_key_ids = Vec::new();
+        for i in 0..keys_per_signer {
+            signer_key_ids.push(keys_per_signer * signer_id + i + 1);
+        }
+        signer_key_ids
+    }
+
+    fn create_public_keys(signer_private_keys: &Vec<Scalar>, keys_per_signer: u32) -> PublicKeys {
+        let signer_id_keys = (0..signer_private_keys.len())
+            .map(|i| {
+                (
+                    (i + 1) as u32,
+                    ecdsa::PublicKey::new(&signer_private_keys[i]).unwrap(),
+                )
+            })
+            .collect::<HashMap<u32, ecdsa::PublicKey>>();
+
+        let mut key_ids = HashMap::new();
+        for (signer_id, signer_key) in &signer_id_keys {
+            for i in 0..keys_per_signer {
+                key_ids.insert(3 * signer_id - i, signer_key.clone());
+            }
+        }
```

Comment:
Nit: Same here, this is slightly cleaner (also avoiding vector indexing by using `.enumerate`)
```suggestion
        let signer_id_keys: HashMap<u32, ecdsa::PublicKey> = signer_private_keys
            .iter()
            .enumerate()
            .map(|(i, key)| ((i + 1) as u32, ecdsa::PublicKey::new(key).unwrap()))
            .collect();

        let key_ids: HashMap<u32, ecdsa::PublicKey> = signer_id_keys
            .iter()
            .flat_map(|(signer_id, signer_key)| {
                (0..keys_per_signer).map(|i| (3 * *signer_id - i, signer_key.clone()))
            })
            .collect();
```

Reply:
Done.